pkg/cloud/gcp/actuators/machine/actuator.go

```diff
@@ -41,59 +57,154 @@ func NewActuator(params ActuatorParams) *Actuator {
 // Create creates a machine and is invoked by the machine controller.
 func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machine *machinev1.Machine) error {
 	klog.Infof("Creating machine %q", machine.Name)
-	scope, err := newMachineScope(machineScopeParams{
-		machineClient: a.machineClient,
-		coreClient:    a.coreClient,
-		machine:       machine,
-	})
+
+	machineCtx, err := newMachineContext(a.coreClient, machine)
 	if err != nil {
-		return fmt.Errorf(scopeFailFmt, machine.Name, err)
+		return err
+	}
+
+	instance, err := instanceGet(machineCtx, machine.Name)
+	if err != nil {
+		return err
+	}
+	if instance == nil {
+		instance = instanceFromMachineContext(machineCtx, machine.Name)
+		operation, err := machineCtx.computeService.InstancesInsert(machineCtx.projectID, machineCtx.providerSpec.Zone, instance)
+		if err != nil {
+			return err
+		}
+		klog.Infof("Creating instance %s/%s/%s using operation #%v", machineCtx.projectID, machineCtx.providerSpec.Zone, machine.Name, operation.Id)
+		// this forces further controller Reconciliation
+		machine.Spec.ProviderID = &machineCtx.providerID
+		_, err = a.machineClient.Machines(machine.Namespace).Update(machine)
+		return err
```

Comment:
Since we are dropping polling this won't reflect the real state of the instance in the cloud since we are returning true regardless the operation status.
Even if a new reconciliation is triggered by `a.machineClient.Machines(machine.Namespace).Update(machine)`  It's not guaranteed that then `Exist()`->`Update()` sets the real machine state/IPs.
In aws we validate the status returned by the api, that happens to be "pending" so we requeueAfter until the cloud state is "success", then we return true and the requeing loop is broken. We need to put a mechanism to emulate the behaviour above here.